packages/fileimport-service/src/filesApi.js

```diff
@@ -1,22 +1,45 @@
 /* istanbul ignore file */
 'use strict'
+const { ensureError } = require('@speckle/shared')
 const fs = require('fs')
 const path = require('node:path')
-const { stream, fetch } = require('undici')
+const { pipeline } = require('node:stream/promises')
 
 module.exports = {
-  async downloadFile({ fileId, streamId, token, destination }) {
+  async downloadFile({
+    speckleServerUrl,
+    fileId,
+    streamId,
+    token,
+    destination,
+    logger
+  }) {
     fs.mkdirSync(path.dirname(destination), { recursive: true })
-    await stream(
-      `${process.env.SPECKLE_SERVER_URL}/api/stream/${streamId}/blob/${fileId}`,
+    logger.info(
+      { destinationFile: destination },
+      'Downloading file {fileId} from {streamId} to {destinationFile}'
+    )
+    const response = await fetch(
+      `${speckleServerUrl}/api/stream/${streamId}/blob/${fileId}`,
       {
-        opaque: fs.createWriteStream(destination),
         headers: {
           Authorization: `Bearer ${token}`
         }
-      },
-      ({ opaque }) => opaque
+      }
     )
+    const writer = fs.createWriteStream(destination)
+
+    //handle errors
+    writer.on('error', (err) => {
```

Comment:
will the writer still throw an exception if there's errors? or does this `.on` somehow override that default handling and we risk no longer throwing on error?

Reply:
valid question

Reply:
@iainsproat did you test this behavior? its not necessarily clear where exactly this error is gonna bubble out from and whether it needs rethrowing in the first place